src/Agent.Listener/CapabilitiesProvider.cs

```diff
@@ -0,0 +1,291 @@
+                        Path.Combine(System.Environment.GetEnvironmentVariable("HOME") ?? string.Empty, "/Library/Developer/Xamarin/android-sdk-macosx/tools/android")
+                    };
+                _regularCapabilities.Add(new Capability("AndroidSDK", "android", paths));
+            }
+            catch (Exception ex)
+            {
+                Trace.Error(ex);
+            }
+
+            foreach (var cap in _regularCapabilities)
+            {
+                var whichTool = HostContext.GetService<IWhichUtil>();
+                string capPath = whichTool.Which(cap.Tool ?? cap.Name);
+                if (!string.IsNullOrEmpty(capPath))
+                {
+                    capabilities[cap.Name] = capPath;
+                }
+                else if (cap.Paths != null)
+                {
+                    foreach (var path in cap.Paths)
+                    {
+                        if (File.Exists(path))
+                        {
+                            capabilities[cap.Name] = path;
+                            break;
+                        }
+                    }
+                }
+
+                token.ThrowIfCancellationRequested();
+            }
+        }
+
+        private async Task GetToolCapabilities(Dictionary<string, string> capabilities, CancellationToken token)
+        {
+            foreach (var cap in _toolCapabilities)
+            {
+                var whichTool = HostContext.GetService<IWhichUtil>();
+                var toolPath = whichTool.Which(cap.Command);
+                if (string.IsNullOrEmpty(toolPath))
+                {
+                    continue;
+                }
+
+                using (var processInvoker = HostContext.CreateService<IProcessInvoker>())
+                {
+                    string toolOutput = string.Empty;
+                    var outputHandler = new EventHandler<DataReceivedEventArgs>((sender, args) =>
+                    {
+                        if (!string.IsNullOrEmpty(args.Data))
+                        {
+                            toolOutput += args.Data;
+                        }
+                    });
+                    processInvoker.OutputDataReceived += outputHandler;
+                    try
+                    {
+                        await processInvoker.ExecuteAsync(
+                                    workingDirectory: string.Empty,
+                                    fileName: toolPath,
+                                    arguments: cap.CommandArgs,
+                                    environment: null,
+                                    cancellationToken: token);
+                        //toolOutput does not to be a valid file path
+                        if (!string.IsNullOrEmpty(toolOutput))
+                        {
+                            capabilities[cap.Name] = toolOutput;
+                        }
+                    }
+                    catch (OperationCanceledException)
```

Comment:
catch(exception ex) when (!(ex is OperationCanceledException))
{
    Trace.Error(ex)
}